packages/xstate-wallet/src/store/memory-store.ts

```diff
@@ -0,0 +1,302 @@
+import {Observable, fromEvent} from 'rxjs';
+import {filter} from 'rxjs/operators';
+import {EventEmitter} from 'eventemitter3';
+import * as _ from 'lodash';
+import {
+  getChannelId,
+  Outcome,
+  signState,
+  hashState,
+  State as NitroState,
+  getStateSignerAddress
+} from '@statechannels/nitro-protocol';
+
+import {BigNumber, bigNumberify, splitSignature, joinSignature} from 'ethers/utils';
+import {Wallet} from 'ethers';
+
+interface StateVariables {
+  outcome: Outcome;
+  turnNum: BigNumber;
+  appData: string;
+  isFinal: boolean;
+}
+
+interface ChannelConstants {
+  chainId: string;
+  participants: Participant[];
+  channelNonce: BigNumber;
+  appDefinition: string;
+  challengeDuration: BigNumber;
+}
+
+export interface State extends ChannelConstants, StateVariables {
+  channelId: string;
+}
+
+interface ChannelStorage {
```

Comment:
We already have a `ChannelStorage` in the Solidity code, I recommend not re-using that symbol.

https://github.com/statechannels/monorepo/blob/641d7d7987acc67e3ea035e77a1822da93ac92bf/packages/nitro-protocol/contracts/interfaces/IForceMove.sol#L39-L45

```solidity
    struct ChannelStorage {
        uint256 turnNumRecord;
        uint256 finalizesAt;
        bytes32 stateHash; // keccak256(abi.encode(State))
        address challengerAddress;
        bytes32 outcomeHash;
    }
```